p2p/src/authenticated/lookup/actors/peer/actor.rs

```diff
@@ -165,10 +165,11 @@ impl<E: Spawner + Clock + ReasonablyRealtime + Rng + CryptoRng + Metrics, C: Pub
                         .map_err(Error::ReceiveFailed)?;
 
                     // Parse the message
-                    let msg = match types::Message::decode_cfg(msg, &(..).into()) {
+                    let max_data_length = msg.len(); // apply loose bound to data read to prevent memory exhaustion
```

Comment:
Without this bound, we'll pre-allocate whatever the peer provides as the length of the bytes (even if we can't actually read that much).